src/hotspot/share/opto/loopopts.cpp

```diff
@@ -218,6 +229,23 @@ Node* PhaseIdealLoop::split_thru_phi(Node* n, Node* region, int policy) {
   return phi;
 }
 
+// Check if Node 'x' has moved to inner loop compared to Node 'n'
+bool PhaseIdealLoop::has_moved_to_inner_loop(Node* n, Node* region, Node* x) {
+  assert(region->is_Loop(), "region should be a loop");
+  IdealLoopTree* n_loop_tree = get_loop(region);
+  for (uint j = 1; j < n->req(); j++) {
```

Comment:
Why do you need this loop at all? Wouldn't it be sufficient to just check `x_loop_tree->is_member(n_loop_tree)`?

Reply:
you are right! I changed it